pkgs/tools/networking/smartdns/default.nix

```diff
@@ -0,0 +1,55 @@
+
+{ stdenv, fetchFromGitHub, openssl, fetchpatch }:
+
+stdenv.mkDerivation rec {
+  pname = "smartdns";
+  version = "30";
+  src = fetchFromGitHub {
+    owner = "pymumu";
+    repo = pname;
+    rev = "Release${version}";
+    sha256 = "1zj1ajvdnpmdal87j7fyn0xdd0ydjii1nvgwgacv2j311b6fdrk7";
+  };
+
+  buildInputs = [ openssl ];
+
+  # TODO: A patch is needed by now since the upstream doesn't follow the FHS guideline. It should not be the case in the future.
+  patches = [
+    (fetchpatch {
+      url =
+        "https://git.archlinux.org/svntogit/community.git/plain/trunk/systemd.patch?h=packages/smartdns&id=c15e81742c93e24c47de59a956f1d0d0f7156f0a";
+      name = "systemd.nix";
+      sha256 = "0jxzsgxz6y1a4mvwk6d5carxv4nxsswpfs1a6jkplzy2gr0g8281";
+    })
+  ];
+
+  patchFlags = [ "-p1" "systemd/smartdns.service" ];
+
+  postPatch =
+    "substituteInPlace systemd/smartdns.service --replace /usr/bin/smartdns $out/bin/smartdns";
```

Comment:
```suggestion
  postPatch = ''
    substituteInPlace systemd/smartdns.service --replace /usr/bin/smartdns $out/bin/smartdns
  '';
```

Reply:
Why? shouldn't I keep it in one-liner form?

Reply:
This is a convention to make the diffs clean when adding new commands to the `postPatch` section.

Reply:
Also it will avoid the unidiomatic line break between `=` and `"`.

Reply:
OK